streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamAggregate.java

```diff
@@ -118,10 +120,13 @@ public void process(final Record<KIn, VIn> record) {
 
             newAgg = aggregator.apply(record.key(), record.value(), oldAgg);
 
-            store.put(record.key(), newAgg, newTimestamp);
-            tupleForwarder.maybeForward(
-                record.withValue(new Change<>(newAgg, sendOldValues ? oldAgg : null))
-                    .withTimestamp(newTimestamp));
+            final long putReturnCode = store.put(record.key(), newAgg, newTimestamp);
+            // if not put to store, do not forward downstream either
+            if (putReturnCode != PUT_RETURN_CODE_NOT_PUT) {
+                tupleForwarder.maybeForward(
+                    record.withValue(new Change<>(newAgg, sendOldValues ? oldAgg : null, putReturnCode == PUT_RETURN_CODE_IS_LATEST))
```

Comment:
`putReturnCode == PUT_RETURN_CODE_IS_LATEST` should always be `true` ?

Reply:
Same as above -- I'll simplify this if it's correct to.